tutorial/SparkRedshiftTutorial.scala

```diff
@@ -0,0 +1,172 @@
+import java.io.File
+import org.apache.spark.sql.types.{ StructType, StructField, 
+                                    StringType, IntegerType, 
+                                    LongType, DecimalType }
+
+/**
+ * Source code accompanying the spark-redshift tutorial. 
+ * The following parameters need to be passed
+ * 1. AWS Access Key
+ * 2. AWS Secret Access Key
+ * 3. Redshift Database Name
+ * 4. Redshift UserId
+ * 5. Redshift Password
+ * 6. Redshift URL (Ex. swredshift.czac2vcs84ci.us-east-1.redshift.amazonaws.com:5439
+ * 
+ */
+object SparkRedshiftTutorial {
+  /*
+   * For Windows Users only
+   * 1. Download contents from link 
+   *      https://github.com/srccodes/hadoop-common-2.2.0-bin/archive/master.zip
+   * 2. Unzip the file in step 1 into your %HADOOP_HOME%/bin. 
+   *        In the example below  %HADOOP_HOME%=C:/MySoftware/hdphome
+   * 3. After step 2 %HADOOP_HOME/bin should contain winutils.exe
+   * 
+   */   
+	if (System.getProperty("os.name").startsWith("Windows")) {
+		val hdphome: File = new File("C:/MySoftware/hdphome");
+		System.setProperty("hadoop.home.dir", hdphome.getAbsolutePath().toString())
+	}
+
+	def main(args: Array[String]): Unit = {
+		val awsAccessKey = args(0)
+		val awsSecretKey = args(1)
+		val rsDbName = args(2)
+		val rsUser = args(3)
+		val rsPassword = args(4)
+    /** Sample Redshift URL is swredshift.czac2vcs84ci.us-east-1.redshift.amazonaws.com:5439 */
+		val rsURL = args(5)  
+		val jdbcURL = s"""jdbc:redshift://$rsURL/$rsDbName?user=$rsUser&password=$rsPassword"""
+    println(jdbcURL)
+		val sc = new SparkContext(new SparkConf().setAppName("SparkSQL").setMaster("local"))
+
+		val tempS3Dir = "s3n://redshift-spark/temp/"
+		sc.hadoopConfiguration.set("fs.s3n.awsAccessKeyId", awsAccessKey)
+		sc.hadoopConfiguration.set("fs.s3n.awsSecretAccessKey", awsSecretKey)
+
+		val sqlContext = new org.apache.spark.sql.SQLContext(sc)
+
+		import sqlContext.implicits._
+    /** Load from a table */
+		val eventsDF = sqlContext.read
+			.format("com.databricks.spark.redshift")
+			.option("url", jdbcURL) 
+			.option("tempdir", tempS3Dir) 
+			.option("dbtable", "event") 
+			.load()
+		eventsDF.show()
+		eventsDF.printSchema()
+
+    /** Load from a query */
+		val salesQuery = """select salesid,listid,sellerid,buyerid,
+                        eventid,dateid,qtysold,pricepaid,commission 
+                        from sales 
+                        order by saletime desc LIMIT 10000"""
+		val salesDF = sqlContext.read
+			.format("com.databricks.spark.redshift")
+			.option("url", jdbcURL)
+			.option("tempdir", tempS3Dir)
+			.option("query", salesQuery)
+			.load()
+		salesDF.show()
+
+		val eventQuery = "select * from event"
+		val eventDF = sqlContext.read
+			.format("com.databricks.spark.redshift")
+			.option("url", jdbcURL) 
+			.option("tempdir", tempS3Dir)
+			.option("query", eventQuery)
+			.load()
+
+    /** Register 'event' table as 'myevent' in the Spark Environment */
```

Comment:
Also, what is "Spark Environment" here? It seems more straightforward to explain that this is registering a temporary table so that it can be queried via the `sqlContext.sql` interface, etc.